src/parser/type_checking.c

```diff
@@ -22979,11 +22980,18 @@ pt_common_collation (PT_COLL_INFER * arg1_coll_infer, PT_COLL_INFER * arg2_coll_
 	{
 	  goto error;
 	}
-      *common_coll = arg2_coll_infer->coll_id;
+      if (arg2_coll_infer->coll_id > arg1_coll_infer->coll_id)
+	{
+	  *common_coll = arg1_coll_infer->coll_id;
+	}
+      else
+	{
+	  *common_coll = arg2_coll_infer->coll_id;
+	}
       *common_cs = arg2_coll_infer->codeset;
```

Comment:
Common_coll and common_cs are being allocated from different data(arg1, arg2). is it okay?

Reply:
I fixed it.